python/cuml/tsa/auto_arima.pyx

```diff
@@ -306,7 +306,7 @@ class AutoARIMA(Base):
                         break
                 else:  # (when the for loop reaches its end naturally)
                     # The remaining series are assigned the max possible d
-                    data_dD[d_options[-1]] = (data_temp, id_temp)
+                    data_dD[(d_options[-1], D_)] = (data_temp, id_temp)
```

Comment:
Could we add small pytests that hit this and all code paths if possible?

Reply:
Agreed. I opened the issue #2652 about adding an integration test (and described in the issue why it is challenging to test auto-ARIMA properly).